client-ui/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -13,200 +13,132 @@
-          this.totalCount = executionEvents.totalCount
+        .pipe(filter(_.identity))
+        .subscribe(executionEvents => {
+          console.log("received data", executionEvents)
+          this.executionEvents = executionEvents
+          this.loadingIndicator.deactivate()
         })
     )
   }
 
-  ngAfterViewInit(): void {
-
-    this.subscriptions.push(
-      this.table.page.pipe(
-        tap(_ => this.loading = true),
-      ).subscribe(
-        page => {
-          this.queryParams = { ...this.queryParams, offset: page.offset }
-          this.applyFilters()
-        }
-      )
-    )
-
-    this.subscriptions.push(
-      this.table.sort.pipe(
-        tap(_ => this.loading = true),
-        map(event => event.sorts[0]),
-      ).subscribe(
-        sort => {
-          this.queryParams = { ...this.queryParams, offset: 0, sortName: sort.prop, sortDirection: sort.dir }
-          this.applyFilters()
-        }
-      )
-    )
-
-    this.subscriptions.push(
-      fromEvent<any>(this.searchInput.nativeElement, 'keyup')
-        .pipe(
-          tap(_ => this.loading = true),
-          map(event => event.target.value),
-          debounceTime(400)
-        ).subscribe(
-          searchTerm => {
-            this.queryParams = { ...this.queryParams, offset: 0, searchTerm: searchTerm }
-            this.applyFilters()
-          }
-        )
-    )
-
-  }
-
-  private applyFilters(): void {
-    if (this.useDateRange) {
-      this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
-    }
-    this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
-    this.loading = false
+  private loadExecutionEvents(dashboardState): void {
+    const dateRange = dashboardState.filters.dateRange
+    this.store.dispatch(new ExecutionEventsActions.Get(
+      {
+        asAtTime: this.asAtTime,
+        timestampStart: dateRange && dateRange[0],
+        timestampEnd: dateRange && dateRange[1],
+        offset: dashboardState.pagination.page,
+        sortName: dashboardState.sort.field,
+        sortDirection: dashboardState.sort.direction,
+        searchTerm: dashboardState.filters.searchQuery,
+      }))
   }
 
-  public onUseDateRangeChange(value): void {
-    this.useDateRange = value
-    this.queryParams = !this.useDateRange && {}
-    this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(s => s.unsubscribe())
   }
 
   public onSelect(event): void {
     const executionEventId = event.selected[0].executionEventId
     const params = {} as RouterStateUrl
-    params.queryParams = { "executionEventId": executionEventId }
+    params.queryParams = {"executionEventId": executionEventId}
     params.url = "/app/lineage-overview/"
     this.store.dispatch(new RouterAction.Go(params))
   }
 
+  public onFilterByDateSwitchToggle(value): void {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected parameter: 'value' to have a typedef](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)